R/util.R

```diff
@@ -33,16 +33,44 @@ pandoc_output_ext <- function(ext, to, input) {
   paste0(".", to)
 }
 
-pkg_file <- function(...) {
-  system.file(..., package = "rmarkdown")
-}
-
-pkg_file_arg <- function(...) {
-  pandoc_path_arg(pkg_file(...))
-}
-
-pkg_file_lua <- function(...) {
-  pkg_file("rmd", "lua", ...)
+pkg_file <- function(..., package = "rmarkdown") {
+  system.file(..., package = package)
+}
+
+pkg_file_arg <- function(..., package = "rmarkdown") {
+  pandoc_path_arg(pkg_file(..., package = package))
+}
+
+#' Get Lua filters full paths in a package
+#'
+#' Lua filters stored in a source package in \code{inst/rmarkdown/lua} will installed in
+#' \code{rmarkdown/lua} in the package path. This function finds the full path of the
+#' Lua filters on the system for installed packages.
+#'
+#'
+#' @param filters Character vector of filename for the Lua filter to retrieve
+#'   in \code{rmarkdown/lua} folder of the package. If \code{NULL}, all the
+#'   included lua filter will be returned. The extension can be ommited.
+#' @param package Package name in which to look for the filters.
+#' @return Character vector of absolute file paths for the Lua filter from the
+#'   package. If a filter is not found, it does not error but returns an empty
+#'   path. The path will already be escaped correctly using
+#'   \code{\link{pandoc_path_arg}} to be used by Pandoc.
+#' @export
+#' @examples
+#' # get all filters
+#' pkg_file_lua(package = "rmarkdown")
+#' # get a specific filter
+#' pkg_file_lua("pagebreak", package = "rmarkdown")
+#' pkg_file_lua(c("pagebreak", "latex_divs"), package = "rmarkdown")
+#' pkg_file_lua("pagebreak.lua", package = "rmarkdown")
+pkg_file_lua <- function(filters = NULL, package = NULL) {
```

Comment:
We can make our own life easier by giving the `package` a default:

```suggestion
pkg_file_lua <- function(filters = NULL, package = "rmarkdown") {
```

Reply:
I wanted to do that at first, but as the function is now exported I did not because I thought it would strange for the user to have this function default to the filter from inside **rmarkdown**. I may be overthinking it ?

Reply:
It's also important to make our own code simpler :) In the current PR, you had to use `package = "rmarkdown"` in many places. If this is the default, you wouldn't need to add these `package` arguments repetitively.

This function belongs to the **rmarkdown** package, and I don't feel it's strange that the package name defaults to **rmarkdown**. For other package authors, they can easily write their own wrapper function, e.g.,

```r
pkg_file_lua <- function(...) {
  rmarkdown::pkg_file_lua(..., package = "foo")
}
```

Reply:
I see. It make complete sense. I'll remember that. 
Done f8da23b